--- mdbtools-0.5/src/extras/Makefile.am.orig	2001-04-12 01:33:19.000000000 +0200
+++ mdbtools-0.5/src/extras/Makefile.am	2003-04-20 18:20:11.000000000 +0200
@@ -1,5 +1,5 @@
 bin_PROGRAMS	=	mdb-dump
 mdb_dump_SOURCES = mdb-dump.c mdbsupport.c
-INCLUDES	=	-I$(top_srcdir)/include `glib-config --cflags`
+INCLUDES	=	-I$(top_srcdir)/include $(GLIB_CFLAGS)
 LDADD	=	../libmdb/libmdb.la 
-LIBS	=	`glib-config --libs`
+LIBS	=	$(GLIB_LIBS)
--- mdbtools-0.5/src/odbc/Makefile.am.orig	2002-12-27 21:31:30.000000000 +0100
+++ mdbtools-0.5/src/odbc/Makefile.am	2003-04-20 18:19:14.000000000 +0200
@@ -6,13 +6,13 @@
 				data.c like.c table.c dump.c file.c mem.c
 bin_PROGRAMS	=	unittest
 lib_LTLIBRARIES	=	libmdbodbc.la
-INCLUDES	=	-I ../../include `glib-config --cflags`
+INCLUDES	=	-I ../../include $(GLIB_CFLAGS)
 libmdbodbc_la_SOURCES=	odbc.c connectparams.c
 libmdbodbc_la_LIBADD=	$(patsubst %, $(SQLDIR)/%, \
 				$(patsubst %.c, %.lo, $(SQLSOURCES))) \
 				$(patsubst %, $(MDBDIR)/%, \
-				$(patsubst %.c, %.lo, $(MDBSOURCES))) -lglib
-LIBS	=	@LEXLIB@ `glib-config --libs`
+				$(patsubst %.c, %.lo, $(MDBSOURCES)))
+LIBS	=	@LEXLIB@ $(GLIB_LIBS)
 unittest_LDADD	=	libmdbodbc.la ../libmdb/libmdb.la ../sql/libmdbsql.la
 
 ## Need blank statement to avoid compiling odbc.c
--- mdbtools-0.5/configure.in.orig	2003-01-12 23:59:41.000000000 +0100
+++ mdbtools-0.5/configure.in	2003-04-20 18:24:56.000000000 +0200
@@ -103,7 +103,6 @@
 fi
 
 AC_SUBST([OPTDIRS])
-AC_CONFIG_FILES([src/Makefile])
 
 dnl Checks for library functions.
 
@@ -122,4 +122,4 @@
 
 AC_SUBST(READLINE_LIBS)
 
-AC_OUTPUT(src/util/Makefile src/extras/Makefile Makefile include/Makefile src/Makefile src/libmdb/Makefile include/Makefile src/sql/Makefile src/odbc/Makefile doc/Makefile src/gmdb2/Makefile src/gmdb2/gladefiles/Makefile src/gmdb2/pixmaps/Makefile src/gmdb2/help/Makefile src/gmdb2/help/C/Makefile mdbtools.spec)
+AC_OUTPUT(src/util/Makefile src/extras/Makefile Makefile include/Makefile src/Makefile src/libmdb/Makefile src/sql/Makefile src/odbc/Makefile doc/Makefile src/gmdb2/Makefile src/gmdb2/gladefiles/Makefile src/gmdb2/pixmaps/Makefile src/gmdb2/help/Makefile src/gmdb2/help/C/Makefile mdbtools.spec)
