--- mdbtools-0.6pre1/src/odbc/Makefile.am.orig	2006-10-25 00:56:42.000000000 +0000
+++ mdbtools-0.6pre1/src/odbc/Makefile.am	2006-10-25 00:56:51.000000000 +0000
@@ -14,7 +14,7 @@
 libmdbodbc_la_LIBADD=	$(patsubst %, $(SQLDIR)/%, \
 				$(patsubst %.c, %.lo, $(SQLSOURCES))) \
 				$(patsubst %, $(MDBDIR)/%, \
-				$(patsubst %.c, %.lo, $(MDBSOURCES)))
+				$(patsubst %.c, %.lo, $(MDBSOURCES))) -lm
 LIBS	=	@LEXLIB@ $(GLIB_LIBS) $(ODBC_LIBS)
 unittest_LDADD	=	libmdbodbc.la ../libmdb/libmdb.la ../sql/libmdbsql.la
 
