--- mdbtools-0.5/src/odbc/Makefile.am~	2004-07-11 13:01:16.731636064 +0200
+++ mdbtools-0.5/src/odbc/Makefile.am	2004-07-11 13:01:39.867118936 +0200
@@ -11,7 +11,7 @@
 libmdbodbc_la_LIBADD=	$(patsubst %, $(SQLDIR)/%, \
 				$(patsubst %.c, %.lo, $(SQLSOURCES))) \
 				$(patsubst %, $(MDBDIR)/%, \
-				$(patsubst %.c, %.lo, $(MDBSOURCES))) -lglib
+				$(patsubst %.c, %.lo, $(MDBSOURCES)))
 LIBS	=	@LEXLIB@ $(GLIB_LIBS)
 unittest_LDADD	=	libmdbodbc.la ../libmdb/libmdb.la ../sql/libmdbsql.la
 
