diff -urN mdbtools-0.6pre1/libmdb.pc.in mdbtools-0.6pre1.new//libmdb.pc.in
--- mdbtools-0.6pre1/libmdb.pc.in	2004-05-05 03:38:50.000000000 +0200
+++ mdbtools-0.6pre1.new//libmdb.pc.in	2010-11-03 11:53:18.026752506 +0100
@@ -12,4 +12,4 @@
 Requires: glib-2.0
 Version: @VERSION@
 Libs: -L${libdir} -lmdb
-Cflags:
+Cflags: -I${includedir}
diff -urN mdbtools-0.6pre1/libmdbsql.pc.in mdbtools-0.6pre1.new//libmdbsql.pc.in
--- mdbtools-0.6pre1/libmdbsql.pc.in	2004-05-05 03:38:50.000000000 +0200
+++ mdbtools-0.6pre1.new//libmdbsql.pc.in	2010-11-03 11:54:48.206752506 +0100
@@ -12,5 +12,5 @@
 Requires: libmdb
 Version: @VERSION@
 Libs: -L${libdir} -lmdbsql
-Cflags:
+Cflags: -I${includedir}
 
